### PR TITLE
ng_sixlowpan: fix IPHC dispatch check

### LINKED ISSUE #3411: Fix 6LoWPAN IPv6 Header Compression check
The following log is produced when sending the attached PCAP (download the image and change the extension from png to pcapng) to RIOT. It was captured during testing with this guide:
https://github.com/RIOT-OS/RIOT/wiki/How-to-install-6LoWPAN-Linux-Kernel-on-Raspberry-Pi

```
2015-07-15 10:58:20,934 - INFO # nomac: sending pkt 0x200019f0 to PID 5
2015-07-15 10:58:20,937 - INFO # 6lo: NG_NETDEV_MSG_TYPE_RCV received
2015-07-15 10:58:20,940 - INFO # 6lo: dispatch 7b ... is not supported
2015-07-15 10:58:20,943 - INFO # 6lo: waiting for incoming message.
```

````
> sha1sum raspberry_pi_linux_6lowpan_header_compression.pcapng 
d0b2bdf99b671a3fa6593abcb9436bcd308a36d8  raspberry_pi_linux_6lowpan_header_compression.pcapng
```
![raspberry_pi_linux_6lowpan_header_compression](https://cloud.githubusercontent.com/assets/254701/8695213/0afff71c-2ae4-11e5-9228-7cbd66132c76.png)

### PR DESCRIPTION
**Pull request by authmillenon**
Fixes #3411

### CODE REVIEW

File: sys/net/network_layer/ng_sixlowpan/ng_sixlowpan.c [hunk 1]
```diff
@@ -121,15 +121,8 @@ static void _receive(ng_pktsnip_t *pkt)
...
-    else {
-        DEBUG("6lo: dispatch %02x ... is not supported\n",
-              dispatch[0]);
-        ng_pktbuf_release(pkt);
-        return;
-    }
-
 #ifdef MODULE_NG_SIXLOWPAN_IPHC
-    if (ng_sixlowpan_iphc_is(payload->data)) {
+    else if (ng_sixlowpan_iphc_is(payload->data)) {
```
Comment (OlegHahm) [inline sys/net/network_layer/ng_sixlowpan/ng_sixlowpan.c]: why not using `dispatch` here?

**Comment by authmillenon** [comment]
Addressed comments

**Comment by OlegHahm** [comment]
ACK, please squash!

**Comment by authmillenon** [comment]
Squashed

**Comment by gebart** [comment]
@phiros @authmillenon Why didn't Travis build the ng_networking example in this case?

**Comment by authmillenon** [comment]
Because it does not include `ng_sixlowpan_iphc` right now, I guess.

**Comment by gebart** [comment]
but the change is in the file  sys/net/network_layer/ng_sixlowpan/ng_sixlowpan.c ?

**Comment by gebart** [comment]
isn't sixlowpan pulled in as a dep when building for platforms with 802.15.4 transceivers?

**Comment by gebart** [comment]
I think the iot-lab_M3 would have been rebuilt at least since that uses 6lowpan

**Comment by authmillenon** [comment]
Yes and after reviewing `dist/tools/compile_test/compile_test.py` I see whats happening: It only checks the change set for a specific board, if there were changes to the board's or cpu's file in the `boards/` or `cpu/` directory specifically (https://github.com/RIOT-OS/RIOT/blob/master/dist/tools/compile_test/compile_test.py#L124). So it just checks for native in this case (where 6LoWPAN is not included). Can someone open a PR/issue to fix that? I'm currently not finding the time for that :(

**Comment by gebart** [comment]
@phiros Do you know how to improve compile_test.py to fix the problem?

**Comment by authmillenon** [comment]
This is however unrelated to this PR, so I merge.

**Pull request merged**